sdcm/utils/cdc/gce_builder.py

```diff
@@ -0,0 +1,250 @@
+        """
+        # The template describes the size and source image of the boot disk
+        # to attach to the instance.
+        disk = compute_v1.AttachedDisk()
+        initialize_params = compute_v1.AttachedDiskInitializeParams()
+        initialize_params.source_image = (
+            self.runner.image.extra.get('selfLink')
+        )
+        initialize_params.disk_type = 'pd-standard'
+        initialize_params.disk_size_gb = 80
+        disk.initialize_params = initialize_params
+        disk.auto_delete = True
+        disk.boot = True
+
+        # The template connects the instance to the `default` network,
+        # without specifying a subnetwork.
+        network_interface = compute_v1.NetworkInterface()
+        network_interface.network = self.region.network.self_link
+
+        # The template lets the instance use an external IP address.
+        access_config = compute_v1.AccessConfig()
+        access_config.name = "External NAT"
+        access_config.type_ = "ONE_TO_ONE_NAT"
+        access_config.network_tier = "PREMIUM"
+        network_interface.access_configs = [access_config]
+
+        template = compute_v1.InstanceTemplate()
+        template.name = self.name
+        template.properties.disks = [disk]
+        template.properties.machine_type = "e2-medium"
+        template.properties.network_interfaces = [network_interface]
+
+        metadata = compute_v1.Metadata()
+        metadata.items += [{"key": "RunByUser", "value": "QA"}]
+        metadata.items += [{"key": "NodeType", "value": "builder"}]
+        metadata.items += [{"key": "keep", "value": "alive"}]
+
+        template.properties.metadata = metadata
+        template_client = compute_v1.InstanceTemplatesClient(credentials=self.credentials)
+        try:
+            operation = template_client.insert(
+                project=self.region.project, instance_template_resource=template
+            )
+
+            self.wait_for_extended_operation(operation, "instance template creation")
+        except google.api_core.exceptions.Conflict:
+            pass
+
+        return template_client.get(project=self.region.project, instance_template=self.name)
+
+    def configure(self):
+        self.create_launch_template()
+        self.add_cloud_configuration_to_jenkins()
+
+    @classmethod
+    def configure_in_all_region(cls, regions=None):
+        for project in SUPPORTED_PROJECTS:
+            os.environ['SCT_GCE_PROJECT'] = project
+
+            regions = regions or ['us-east1', 'us-west1']
+            for region_name in regions:
+                region = cls(GceRegion(region_name))
+                region.configure()
+
+
+configure_jenkins_builders: click.Command
+
+
+@click.command("configure-jenkins-builders", help="Configure all required jenkins builders for SCT")
+@click.option("-r", "--regions", type=CloudRegion(cloud_provider='gce'),
+              default=[], help="Cloud regions", multiple=True)
+def configure_jenkins_builders(regions):
```

Comment:
still need to align this code with the AWS configure_jenkins_builders,
to make it one command ?  or multiple commands ?

Reply:
this could be perhaps an interface for each different backend, so the caller can just use the function regardless the backend in use.. similar to the unified provision.. and @KnifeyMoloko is still pending to finish the GCE part

Reply:
yes that the intended direction, once I have it working as I expect

Any this is used only when configuration need to be altered, hopefully it won't happen that often.

Reply:
moved it to be shared with the AWS command